### PR TITLE
Fix powershell transport

### PR DESCRIPTION
**Pull request by Mistuke**
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/#conventions).
* [ ] Any changes that could be relevant to users have been recorded in the changelog.
* [ ] The documentation has been updated, if necessary.
* [ ] If the change is docs-only, `[ci skip]` is used to avoid triggering the build bots.

This fixes the powershell transport protocol. The reason for the breakage is because certain headers like
"Range" are protected in the .NET framework https://msdn.microsoft.com/en-us/library/system.net.httpwebrequest(v=vs.110).aspx which means you can't set them purely as metadata and need to go through the proper getter/setters.

### CODE REVIEW

**Comment by 23Skidoo** [comment]
We should add some AppVeyor tests so that we get notified if this breaks again. Note that AppVeyor has `curl` in `PATH` and so chooses the `curl` transport by default. You can force the PowerShell transport with the `--http-transport` option.

**Comment by hvr** [comment]
@23Skidoo is it enough to change the `cabal update` line in the `appveyor.yml` script to say `cabal --http-transport=powershell update -v`, or did you have something more extensive in mind?

**Comment by 23Skidoo** [comment]
If @Mistuke can come up with more extensive tests, that'd be appreciated, but I won't ask for anything more.

File: cabal-install/Distribution/Client/HttpUtils.hs [hunk 1]
```diff
@@ -545,16 +545,37 @@ powershellTransport prog =
...
+          [ "$response = $request.GetResponse()"
+          , "$responseStream = $response.GetResponseStream()"
+          , "$buffer = new-object byte[] 10KB"
+          , "$count = $responseStream.Read($buffer, 0, $buffer.length)"
+          , "while ($count -gt 0)"
+          , "{"
+          , "    $targetStream.Write($buffer, 0, $count)"
+          , "    $count = $responseStream.Read($buffer, 0, $buffer.length)"
+          , "}"
           , "Write-Host \"200\";"
```
Comment (hvr) [inline cabal-install/Distribution/Client/HttpUtils.hs]: The unconditional "200" seems wrong

**Comment by 23Skidoo** [comment]
Merged, thanks!

**Pull request merged**